src/lib/cloud-provider.js

```diff
@@ -148,7 +168,7 @@ class CloudProvider {
         if (this.connection &&
             this.connection.readyState !== WebSocket.CLOSING &&
             this.connection.readyState !== WebSocket.CLOSED) {
-
+            this.connection.onclose = () => {}; // Remove close listener to prevent reconnect
```

Comment:
Hmmm what's the benefit of having the reconnect logic happen in onClose instead of onError? Is it that onError is too soon to attempt a reconnect?

Reply:
Also maybe keep the log here so that we can be sure that the connection actually closed when we want it to.

Reply:
Yeah, error is called before close. They always go together, but error first. Also error is not always called when the socket is closed by the server.

Reply:
Not resolving this conversation yet b/c I still think we should continue to log closing a connection, when the close is explicitly requested.